### PR TITLE
TASK: Remove obsolete `aggregate` as Neos Ui handles nodeMove strategy now via special configuration

### PR DESCRIPTION
**Pull request by pKallert**
As the Neos ui now uses `options.moveNodeStrategy` to determine whether to move all nodes or just on dimension: https://github.com/neos/neos-ui/pull/3876
And we discussed that we DONT want to make that dependant on `aggregate: true` in the core like in 8.3

And the fact that `aggregate` is now fully unused and its description and docs are outdated we removed the unused `aggregate` flag on NodeTypes.
It was originally introduced via https://github.com/neos/neos-development-collection/commit/defa57257ff293e849c024813f44734041411ff3
We decided against using the aggregate flag to determine scatter or gather node move behaviour like in Neos 8 because the naming aggregate might be confusing now.

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Command/MoveNodeAggregate.php [hunk 1]
```diff
@@ -72,12 +72,12 @@ private function __construct(
      * @param WorkspaceName $workspaceName The workspace in which the move operation is to be performed
      * @param DimensionSpacePoint $dimensionSpacePoint This is one of the *covered* dimension space points of the node aggregate and not necessarily one of the occupied ones. This allows us to move virtual specializations only when using the scatter strategy
      * @param NodeAggregateId $nodeAggregateId The id of the node aggregate to move
-     * @param RelationDistributionStrategy $relationDistributionStrategy The relation distribution strategy to be used ({@see RelationDistributionStrategy})
+     * @param RelationDistributionStrategy|null $relationDistributionStrategy The relation distribution strategy to be used ({@see RelationDistributionStrategy})
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Feature/NodeMove/Command/MoveNodeAggregate.php]:
```suggestion
     * @param RelationDistributionStrategy|null $relationDistributionStrategy The relation distribution strategy to be used ({@see RelationDistributionStrategy}). If not specified, is read from NodeTypes.yaml (strategy.moveNode)
```

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php [hunk 2]
```diff
@@ -41,6 +41,17 @@ public static function fromString(?string $serialization): self
             : self::STRATEGY_GATHER_ALL;
     }
 
+    public static function fromName(string $name): self
+    {
+        $name = substr($name, strpos($name, '::') + 2);
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: can we assure here that the stuff before the "::" is alwayas the correct string "RelationDistributionStrategy"?

**Review by skurfuerst — COMMENTED** [review]
nice, only two nitpicks :)

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php (hunk 2 above)
Comment (pKallert) [reply Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: done 👍

**Comment by mhsdesign** [comment]
Huuch i remember having created an issue for this once https://github.com/neos/neos-ui/issues/3587 and funnily there was even a pr from Bernhard both for neos and the ui https://github.com/neos/neos-development-collection/pull/4994 but that seems to be superseded then ...

**Comment by bwaidelich** [comment]
> In order to mimic the behavior of 9.0 we decided to introduce a setting for move strategies on NodeType level.

is that correct? I'm confused as to what this PR is supposed to solve (by just reading the description)

**Comment by pKallert** [comment]
> > In order to mimic the behavior of 9.0 we decided to introduce a setting for move strategies on NodeType level.
> 
> is that correct? I'm confused as to what this PR is supposed to solve (by just reading the description)

Sorry, of course I meant 8.3 😄 

This is for differentiating between document and content nodes when moving nodes. 
For example: When moving a document node in the UI the node should be moved in all dimensions. When moving a content node (like in 8.3) the node should perhaps not be moved in all dimensions. 
This is now a setting that can be set for each node type with the behavior from 8.3 set as the default. 

We also talked about implementing a query window in the UI, but that could get annoying really fast especially if you move nodes very often. This can be further enhanced in future iterations but for now we landed on setting it in the node type. 

There are still some edge cases that I have to have a look at and think about(when I get my local environment running again) to fix the failing test.  
I updated the description accordingly and set the PR to draft since the tests are failing and my local env keeps giving errors

**Comment by bwaidelich** [comment]
Thanks a lot for the great explanation! 🤩

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php [hunk 3]
```diff
@@ -41,6 +41,19 @@ public static function fromString(?string $serialization): self
...
+    public static function fromName(string $name): self
+    {
+        if (str_starts_with($name, 'RelationDistributionStrategy::')) {
+            $name = substr($name, strpos($name, '::') + 2);
+            foreach (self::cases() as $status) {
+                if ($name === $status->name) {
+                    return $status;
+                }
+            }
+        }
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: Why this complex logic, every backed enum has a built-in `from` and `tryFrom`

Comment (pKallert) [reply Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: In this case, we want to get the value from key. 
Enum `->from` gets the key from a value, as far as I understand. So I could pass `gatherSpecializations` and would get `RelationDistributionStrategy::STRATEGY_GATHER_SPECIALIZATIONS`. 
Here, we have `RelationDistributionStrategy::STRATEGY_SCATTER`
Basically we need to cast the string `RelationDistributionStrategy::STRATEGY_GATHER_ALL` into an actual enum. Not sure if there is a way to do it that is not ugly?

File: Neos.Neos/NodeTypes/Mixin/Content.yaml [hunk 4]
```diff
@@ -3,6 +3,8 @@
     'Neos.Neos:Node': true
     'Neos.Neos:Hidable': true
   abstract: true
+  strategy:
+    moveNode: RelationDistributionStrategy::STRATEGY_SCATTER
```
Comment (bwaidelich) [inline Neos.Neos/NodeTypes/Mixin/Content.yaml]: I think _that's_ the root cause. Why not
```suggestion
    moveNode: scatter
```

File: Neos.Neos/NodeTypes/Mixin/Document.yaml [hunk 5]
```diff
@@ -7,6 +7,8 @@
     'Neos.Neos:Hidable': true
   abstract: true
   aggregate: true
+  strategy:
+    moveNode: RelationDistributionStrategy::STRATEGY_GATHER_ALL
```
Comment (bwaidelich) [inline Neos.Neos/NodeTypes/Mixin/Document.yaml]: ..and
```suggestion
    moveNode: gatherAll
```

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php (hunk 3 above)
Comment (bwaidelich) [reply Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: I just realized that we use a different value for name and value.
btw that's IMO a limitation of PHP enums, it would make things much easier if enums without backed type would just serialize to string and have `(try)from` to avoid that confusion.
But, anyways, when using backed enums we should probably always refer to the _value_ instead of the _key_ when (de)serializing the enum

File: Neos.Neos/NodeTypes/Mixin/Content.yaml (hunk 4 above)
Comment (pKallert) [reply Neos.Neos/NodeTypes/Mixin/Content.yaml]: This format is what we decided during the sprint when discussing the new setting. It is a lot clearer than only using the key

Comment (bwaidelich) [reply Neos.Neos/NodeTypes/Mixin/Content.yaml]: The thing is: If it is clearer, we should change the values of those enums to reflect this, i.e. change
```php
case STRATEGY_SCATTER = 'scatter';
```
to 
```php
case STRATEGY_SCATTER = 'STRATEGY_SCATTER';
```

(like we did for many other enums exactly to avoid that confusion).

But without that change, having a different serialization format wether it's in the database (or a JSON representation) from the YAML representation is just error prone in my eyes.

Anyways, I don't want to block this PR of course – feel free to ignore me, just my 2ct :)

File: Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php [hunk 6]
```diff
@@ -41,6 +41,19 @@ public static function fromString(?string $serialization): self
...
+        return self::STRATEGY_GATHER_ALL;
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/Feature/NodeMove/Dto/RelationDistributionStrategy.php]: Sorry, one more addition:
If we decide to keep the value based serialization we should at least fail if referring to an invalid value, rather than falling back to some strategy

File: Neos.ContentRepositoryRegistry/Resources/Private/Schema/NodeTypes.schema.yaml [hunk 7]
```diff
@@ -9,6 +9,11 @@ additionalProperties:
     'final': { type: boolean, description: "If final, this node type cannot be subclassed" }
     'aggregate': { type: boolean, description: "INTERNAL. If aggregate, this node type is something like a Document. If not aggregate, it is seen as part of an aggregate node type - has influence on how e.g. moving of nodes across dimensions work." }
 
+    'strategy':
```
Comment (bwaidelich) [inline Neos.ContentRepositoryRegistry/Resources/Private/Schema/NodeTypes.schema.yaml]: rename to `strategies`?

File: Neos.Neos/NodeTypes/Mixin/Document.yaml [hunk 8]
```diff
@@ -7,6 +7,8 @@
     'Neos.Neos:Hidable': true
   abstract: true
   aggregate: true
```
Comment (mhsdesign) [inline Neos.Neos/NodeTypes/Mixin/Document.yaml]: aggregate is dead and can be removed :)

> this is no longer a supported concept. The relationDistributionStrategy must be used instead for default move behavior

File: Neos.Neos/NodeTypes/Mixin/Document.yaml (hunk 5 above)
Comment (mhsdesign) [inline Neos.Neos/NodeTypes/Mixin/Document.yaml]:
```suggestion
    moveNode
```

could be `relationDistribution`

Comment (mhsdesign) [reply Neos.Neos/NodeTypes/Mixin/Document.yaml]: to make this a little easier, if we put it to options 

```yaml
options:
  relationDistributionStrategy: gatherAll
```

and access it via `Neos\ContentRepository\Core\NodeType\NodeType::getOptions`

**Comment by pKallert** [comment]
To summarize our discussion from yesterday:

The Strategy will be set in the Neos UI depeding on NodeType when a node is moved. 
The setting will be moved to options, not in "strategy"
Since the options are not as important, I can just chose the naming however I want to. 
"aggregate" will be removed

**Comment by mhsdesign** [comment]
Thanks for your adjustments, i also removed the `aggregate=true` part from the documentation and the `NodeType` class.
And as the Neos UI now evaluates alone `options.moveNodeStrategy` i moved the configuration to the Neos Ui package. (and we now use `gatherAll` instead of the ENUM name) :)

**Review by mhsdesign — APPROVED** [review]

**Comment by mhsdesign** [comment]
@kitsunet you said you want to test this? It would be really nice, otherwise ill experiment with it.

**Review by kitsunet — APPROVED** [review]
